### ISSUE TITLE
Harmonize Thrust executable location

### ISSUE THREAD

**Issue by francisbouvier** [breach/thrust#264]
Hi,

Currently Thrust executable location depends on witch binding you use:
- With *python* at `$ENV/lib/python3.x/site-packages/pythrust/vendor/thrust/`
- With *node* at `$ENV/node_modules/node-thrust/vendor/thrust/`
- With *go* at `$HOME/vendor/$OS/$ARCH/$VERSION/`

I think it will be good to harmonize all at `$HOME/.thrust/$VERSION/`

Why:
- as a developer if you use *python* or *node* it's common to have a virtual/local environment by project. And you don't want to download and keep an executable of about 100 Mo for each project.
- for end users it doesn't change anything as the whole app will be bundled.

What do you think ?

**Comment by miketheprogrammer** [comment]
Thrust-Go has a SpawnProvisioner Interface. The SpawnProvisioner controls how to check/download/execute/manage a thrust executable. I may change the default to be more Harmonized however, my goal is that users are not actually using the auto download capabilities, and instead packaging the thrust exe with final installs.

**Comment by francisbouvier** [comment]
@miketheprogrammer 
I agree that for end users we need to provide everything (including the Thrust exe) in a bundle app.

But for developers I believe the best way is to share the Thrust exe.
For the *Go* binding it's already the case, but for the *node* and *python* bindings this is not (except if you `sudo pip install` or `npm -g install` but this not very common).

And it's better to have a consistent approach across all bindings of how the Thrust exe is handled.
